packages/react-native/prepare-android-vendor.sh

```diff
@@ -0,0 +1,132 @@
+
+MODE=$(head -1 "$CONFIG_FILE")
+ARG=$(tail -1 "$CONFIG_FILE")
+
+MAVEN_REPO_DIR="$HOME/.m2/repository/com/bugsnag"
+ARTIFACT_NAMES=(
+    bugsnag-android
+    bugsnag-android-core
+    bugsnag-android-ndk
+    bugsnag-plugin-android-ndk
+    bugsnag-plugin-android-anr
+    bugsnag-plugin-react-native
+    )
+REPO_FILE_VARIANTS=(
+    '-sources.jar'
+    '-sources.jar.asc'
+    '.aar'
+    '.aar.asc'
+    '.module'
+    '.module.asc'
+    '.pom'
+    '.pom.asc'
+    )
+
+sed_in_place() {
+    local script="$1"
+    local file="$2"
+
+    if [[ "$OSTYPE" == linux* ]]; then
+        sed -i "$script" "$file"
+    else
+        sed -i '' "$script" "$file"
+    fi
+}
+
+revendor_from_dir() {
+    local src_dir="$(cd "$1" && pwd)"
+    if [ ! -f "$src_dir/gradlew" ]; then
+        echo "Source directory doesn't look like the bugsnag-android repo: $src_dir"
+        exit 1
+    fi
+
+    local dst_dir="$SCRIPT_DIR/android"
+    local com_dir="$dst_dir/com"
+
+    # artefact version needs to be unique to avoid clash with bugsnag-android artefacts
+    # deployed on mavenCentral/jcenter, append a 'react-native' suffix
+    local bugsnag_android_version=$(cat $src_dir/gradle.properties | grep VERSION_NAME)
+    local bugsnag_version_rn="${bugsnag_android_version#*=}-react-native"
+
+    echo "Clearing local maven repo $MAVEN_REPO_DIR"
+    rm -rf "$MAVEN_REPO_DIR"
+
+    echo "Building local copy of bugsnag-android $bugsnag_version_rn"
+    "$src_dir/gradlew" -p "$src_dir" clean assembleRelease publishToMavenLocal -PVERSION_NAME=$bugsnag_version_rn
+
+    echo "Rebuilding vendor dir $dst_dir"
+    rm -rf "$com_dir"
+    mkdir -p "$com_dir"
+    cp -r "$MAVEN_REPO_DIR" "$com_dir/"
+
+    echo "Recording version"
+    rm -rf "$dst_dir/.bugsnag-android-version"
+    echo $(cd "$src_dir" && git rev-parse HEAD) >> "$dst_dir/.bugsnag-android-version"
+
+    sed_in_place "s/api \"com.bugsnag:bugsnag-android:.*/api \"com.bugsnag:bugsnag-android:$bugsnag_version_rn\"/" "$dst_dir/build.gradle"
+    sed_in_place "s/api \"com.bugsnag:bugsnag-plugin-react-native:.*/api \"com.bugsnag:bugsnag-plugin-react-native:$bugsnag_version_rn\"/" "$dst_dir/build.gradle"
+}
+
+use_bugsnag_version() {
+    local version=$1
+    local dst_dir="$SCRIPT_DIR/android"
+    local bugsnag_dir="$dst_dir/com/bugsnag"
+
+    rm -rf "$bugsnag_dir"
+
+    for name in ${ARTIFACT_NAMES[@]}; do
+        local dstdir="$bugsnag_dir/$name"
+        mkdir -p "$dstdir"
+        pushd "$dstdir"
+        for variant in ${REPO_FILE_VARIANTS[@]}; do
+            curl https://repo1.maven.org/maven2/com/bugsnag/$name/$version/$name-$version$variant >>$name-$version$variant
+        done
+    done
+
+    sed_in_place "s/api \"com.bugsnag:bugsnag-android:.*/api \"com.bugsnag:bugsnag-android:$version\"/" "$dst_dir/build.gradle"
+    sed_in_place "s/api \"com.bugsnag:bugsnag-plugin-react-native:.*/api \"com.bugsnag:bugsnag-plugin-react-native:$version\"/" "$dst_dir/build.gradle"
```

Comment:
This logic (and the other instances of `sed_in_place` in this file) makes the git tree dirty again after running `prepare-android-vendor.sh`. It should be moved in to the `update-android.sh` script because its output needs to be checked in. The stage at which the prepare script is run during a release is too late to make changes to the checked in files.